fenix/app/src/main/java/org/mozilla/fenix/components/metrics/MetricsStorage.kt

```diff
@@ -60,6 +65,7 @@ internal class DefaultMetricsStorage(
     /**
      * Checks local state to see whether the [event] should be sent.
      */
+    @Suppress("CyclomaticComplexMethod")
```

Comment:
Is this suppression still needed with your latest changes? NP if it is, but it does seem like something we could try to improve in a follow-up

Reply:
unfortunately it's still triggering the warning.
I was thinking of doing a follow up where I move some of the conditions to separate methods where it makes sense.